### ISSUE TITLE
Flaky test or race condition?

### ISSUE THREAD

**Issue by anuraaga** [line/armeria#404]
Found this on Travis for a PR that didn't touch this code

```
com.linecorp.armeria.it.metrics.DropwizardMetricsIntegrationTest > normal FAILED
    java.lang.AssertionError: expected:<4> but was:<3>
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:645)
        at org.junit.Assert.assertEquals(Assert.java:631)
        at com.linecorp.armeria.it.metrics.DropwizardMetricsIntegrationTest.normal(DropwizardMetricsIntegrationTest.java:67)
```

**Comment by trustin** [comment]
I saw it as well some time ago from CI but could not reproduce locally. Let me try again soon.

**Comment by trustin** [comment]
No idea why the frequency of this failure increased noticeably, but it's beyond annoying now. Let me fix this today.

**Comment by trustin** [comment]
Seems like this is still happening. Doh!

**Comment by anuraaga** [comment]
Yep had it today

com.linecorp.armeria.it.metrics.DropwizardMetricsIntegrationTest > normal FAILED
    java.lang.AssertionError: expected:<3> but was:<2>
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:645)
        at org.junit.Assert.assertEquals(Assert.java:631)
        at
